### PR TITLE
Refactor repository-wide documentation and automation

### PR DESCRIPTION
**Pull request by FiloSottile**
The following commits were directly pushed to `main` in order to set up the policy bot that this documentation refactor references: https://github.com/C2SP/C2SP/compare/74d8de377ede974dbba612a07ef0df33c4b3fe55...1cb55f864783b4aef5b00bad3405ba9d6feb8ede

### CODE REVIEW

**Review by str4d — COMMENTED** [review]
Reviewed 79467950a387b6867b06a7aaa02a218df173a0f4

File: .github/policy/gen.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	"path/filepath"
+	"sort"
+	"strings"
+	"text/template"
+)
+
+//go:generate go run .
+
+func main() {
+	specs, err := filepath.Glob("../../*.md")
```
[the remark this answers is not in the document]
Comment (FiloSottile) [reply .github/policy/gen.go:15]: Nope, `*` does not match the path separator.

File: .github/README.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+* C2SP decisions are **not based on consensus**. Instead, each spec is developed
+  by its [maintainers], who are responsible for reviewing and accepting changes,
+  just like open source projects. This enables rapid, focused, and opinionated
+  development. Since C2SP produces **specifications, not standards**, technical
+  disagreements can be ultimately be resolved by forking.
+* C2SP specs are **updateable**, and follow [semantic versioning]. Most
+  specifications are expected to start at v0.x.x while in “draft” stage, then
+  stay at v1.x.x for as long as they maintain backwards compatibility, ideally
+  forever.
```
[the remark this answers is not in the document]
Comment (FiloSottile) [reply .github/README.md:16]: Hmm, what is a "material change" though? If it's a requirement we should be clear about its expectations. Is it any new feature? Those come very often in draft state. Is it backwards compatibility breaks? In the early stages it's sometimes extra work to assess the backwards compatibility of changes.

**Comment by FiloSottile** [comment]
Pushed 9d4951834a1c7dfa9172477f719ec06ae78920b0 to main to fix policy-bot.

File: .github/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (str4d) [reply .github/README.md:16]: In my mind, it's a hybrid of the two. Basically, anything that is covered under parts 7 or 8 of SemVer (that for 1.0.0 and above requires bumping either major or minor) would result in a minor bump for 0.x.x versions.

(This is effectively how Rust's application of SemVer works; the left-most non-zero version part is bumped for backwards-incompatible changes.)

[the remark this answers is not in the document]
Comment (FiloSottile) [reply .github/README.md]: I see the argument for incrementing the 0.x.y minor version on breaking changes (part 8 of SemVer), but why do it on non-breaking changes? It's assumed that during the draft stage new things will be added all the time, isn't it?

Proposed new language in [937496e](https://github.com/C2SP/C2SP/pull/93/commits/937496e147918260c4472bf8cedb6ea43b3dec2c).

[the remark this answers is not in the document]
Comment (str4d) [reply .github/README.md]: New language LGTM; if any maintainers have questions about it we can talk to them directly or in a separate issue.

**Review by str4d — APPROVED** [review]
ACK 937496e147918260c4472bf8cedb6ea43b3dec2c

**Pull request merged**